plugin-gradle/build.gradle

```diff
@@ -34,6 +35,28 @@ tasks.withType(Test).configureEach {
 	testLogging.showStandardStreams = true
 }
 
+shadowJar {
+	archiveClassifier.set('')
+	relocate 'org.eclipse.jgit', 'shadow.eclipse.jgit'
+	relocate 'org.eclipse.osgi', 'shadow.eclipse.osgi'
+	relocate 'org.eclipse.equinox', 'shadow.eclipse.equinox'
+	relocate 'org.eclipse.core', 'shadow.eclipse.core'
+	relocate 'org.osgi', 'shadow.org.osgi'
+	relocate 'com.googlecode.concurrenttrees', 'shadow.googlecode.concurrenttrees'
+	relocate 'com.googlecode.javaewah', 'shadow.googlecode.javaewah'
+	relocate 'com.googlecode.javaewah32', 'shadow.googlecode.javaewah32'
+	relocate 'dev.equo.solstice', 'shadow.equo.solstice'
+	relocate 'dev.equo.ide', 'shadow.equo.ide'
+	relocate 'org.apache.commons', 'shadow.apache.commons'
+	relocate 'org.apache.felix', 'shadow.apache.felix'
+	relocate 'org.tukaani.xz', 'shadow.tukaani.xz'
+	relocate 'okhttp3', 'shadow.okhttp3'
+	relocate 'okio', 'shadow.okio'
+	relocate 'org.intellij', 'shadow.intellij'
+	relocate 'org.jetbrains', 'shadow.jetbrains'
+	relocate 'org.slf4j', 'shadow.slf4j'
```

Comment:
Use `relocationPrefix "shadow"` might be enough?

Reply:
(together with `enableRelocation = true`)

I got some issues with loading some of the dependencies, like this:
```
Execution failed for task ':tasks'.
> Could not create task ':spotlessGroovyGradle'.
   > java.io.IOException: Failed to load eclipse groovy formatter: java.lang.RuntimeException: java.lang.ClassNotFoundException: shadow.equo.solstice.p2.P2QueryResult
```

So I backed out of relocating most deps and instead focused on JGit only.

Reply:
I tried this on my fork, it works well. if I missing something? See https://github.com/Goooler/spotless/pull/15.

Reply:
Did you test the Greclipse formatter? I would expect the Eclipse-based formatters to fail.

Reply:
That should be included in Maven tests on CI? Do I need some extra running?

Reply:
I think the Gradle and Maven integration tests are not testing against the shadowed jar. I think you have to publish to maven local to test that. Changing the PR so that at least one of the Gradle or Maven integration tests is testing using the shadowed jar would be a great way to prove me wrong.

I'm willing to merge this without such a test, but only if I can understand how it is working. I would expect the eclipse-based formatters should be "unshadeable", and shading them without changing their packages creates really painful bugs.

Reply:
Yeah, will try to add integration tests for using shadow jar.